pkg/sparrow/targets/remote/gitlab/gitlab.go

```diff
@@ -143,27 +147,37 @@ func (c *client) fetchFile(ctx context.Context, f string) (checks.GlobalTarget,
 // so they may be fetched individually
 func (c *client) fetchFileList(ctx context.Context) ([]string, error) {
 	log := logger.FromContext(ctx)
-	log.DebugContext(ctx, "Fetching file list from gitlab")
+	log.DebugContext(ctx, "Preparing to fetch file list from gitlab")
+
+	rawUrl := fmt.Sprintf("%s/api/v4/projects/%d/repository/tree", c.config.BaseURL, c.config.ProjectID)
+	reqUrl, err := url.Parse(rawUrl)
+	if err != nil {
+		log.ErrorContext(ctx, "Could not parse GitLab API repository URL", "url", rawUrl, "error", err)
+	}
+	query := reqUrl.Query()
+	query.Set("ref", c.config.Branch)
+	query.Set("per_page", strconv.Itoa(paginationPerPage))
```

Comment:
I think we should use [keyset pagination](https://gitlab.devops.telekom.de/caas/microservices/orderportal-2-0/-/issues/2) here instead of offset-based pagination. According to gitlab it's more efficient. Shouldn't be a huge change I think. I think you'd only have to add another query param to it for the initial request, after we can just continue using the Link header.

Reply:
https://docs.gitlab.co.jp/ee/api/rest/index.html#keyset-based-pagination

Yeah, and we should add the `order_by=id&sort=asc` to have some kind of determinist fetching.

Reply:
Additionally, the gitlab token header isn't being set anymore. Am I missing something?

Reply:
Switched to [keyset pagination](https://docs.gitlab.com/ee/api/rest/index.html#keyset-based-pagination) & added sorting parameters.

Reply:
Nice catch @puffitos. Added auth header again & added test coverage